core/src/main/java/org/geysermc/geyser/inventory/updater/AnvilInventoryUpdater.java

```diff
@@ -395,8 +397,14 @@ private boolean isCombining(GeyserItemStack input, GeyserItemStack material) {
         return isEnchantedBook(material) || (input.getJavaId() == material.getJavaId() && hasDurability(input));
     }
 
-    private boolean isRepairing(GeyserItemStack input, GeyserItemStack material) {
-        return input.asItem().isValidRepairItem(material.asItem());
+    private boolean isRepairing(GeyserItemStack input, GeyserItemStack material, GeyserSession session) {
+        HolderSet repairable = input.getComponent(DataComponentType.REPAIRABLE);
+        if (repairable == null) {
+            return false;
+        }
+
+        GeyserHolderSet<Item> set = GeyserHolderSet.convertHolderSet(JavaRegistries.ITEM, repairable);
```

Comment:
In this case, might just be better to have an #is(HolderSet, Item) method so we don't have to create a new object?